src/language/core.md

```diff
@@ -0,0 +1,152 @@
+# The core of hacspec
+
+## Crates and modules
+
+hacspec only supports single-module crates, due to a technical limitation
+of the Rust compiler. Inside this single file, a hacspec program shall always
+start with:
+
+```rust, noplaypen
+use hacspec_lib::*;
+
+// Optional: dependencies on other crates containing hacspec programs
+use foo_dep::*;
```

Comment:
I know that using `foo` is common as placeholder. But I'm defender of using names that mean something. Of course there isn't always something useful to put in. But you can use for example `use other_hacpse_crate::*` or ` fn hacpsec_function` etc.